js/widgets/rhythmruler.js

```diff
@@ -2773,8 +2792,17 @@ function RhythmRuler() {
                 dissectHistory.push([history, drum]);
             }
         }
-
+        
         that._dissectHistory = JSON.parse(JSON.stringify(dissectHistory));
+        
+        var myarray = [];
+        for (var i in dissectHistory) {
+            myarray.push(dissectHistory[i]);
+        }
+        
+        storage.setItem("rhythmhistory", JSON.stringify(myarray));
```

Comment:
I don't think we want to do this using storage. It should be saved to the project (like with the Start Block). (If we use storage, and switch between projects, it will get very messy.)

Reply:
@walterbender  I got your point. Actually, it was a bit easy to implement this way(as done in custom mode widget now). So, I took this approach. But , I was unsure about things to store in the rhythm block.  
Now , I think whatever I stored and  retrieved back from storage,just need to store them in rhythm block(same like start). and retrieve them when block is loaded.